app/models/instructeur.rb

```diff
@@ -145,30 +145,26 @@ def create_trusted_device_token
   def dossiers_id_with_notifications(dossiers)
     dossiers = dossiers.followed_by(self)
 
-    updated_demandes = dossiers
-      .joins(:champs)
+    # Relations passed to #or must be “structurally compatible”, i.e. query the same tables.
+    joined_dossiers = dossiers
+      .left_outer_joins(:champs, :champs_private, :avis, :commentaires)
+
+    updated_demandes = joined_dossiers
       .where('champs.updated_at > follows.demande_seen_at')
 
-    updated_annotations = dossiers
-      .joins(:champs_private)
-      .where('champs.updated_at > follows.annotations_privees_seen_at')
+    # We join `:champs` twice, the second time with `has_many :champs_privates`. ActiveRecord generates the SQL: 'LEFT OUTER JOIN "champs" "champs_privates_dossiers" ON …'. We can then use this `champs_privates_dossiers` alias to disambiguate the table in this WHERE clause.
```

Comment:
TIL: When joining on a `has_many:` relationship whose name doesn’t match the foreign table name (here, `champs_privates` vs `champs`, ActiveRecord gives the joined table an alias in the generated SQL (here, `champs_privates_dossiers`). Otherwise, the following where clause would not work. I noticed it because the tests initially failed with the old where clause.